advanced_server.rb

```diff
@@ -0,0 +1,314 @@
+require 'sinatra'
+require 'octokit'
+require 'json'
+require 'openssl' # Used to verify the webhook signature
+require 'jwt'     # Used to authenticate a GitHub App
+require 'time'    # Used to get ISO 8601 representation of a Time object
+require 'logger'
+
+set :port, 3000
+
+
+# This is template code to create a GitHub App server.
+# You can read more about GitHub Apps here: # https://developer.github.com/apps/
+#
+# On its own, this app does absolutely nothing, except that it can be installed.
+# It's up to you to add fun functionality!
+# You can check out one example in advanced_server.rb.
+#
+# This code is a Sinatra app, for two reasons:
+#   1. Because the app will require a landing page for installation.
+#   2. To easily handle webhook events.
+#
+#
+# Of course, not all apps need to receive and process events!
+# Feel free to rip out the event handling code if you don't need it.
+#
+# Have fun!
+#
+
+class GHAapp < Sinatra::Application
+
+  # !!! DO NOT EVER USE HARD-CODED VALUES IN A REAL APP !!!
```

Comment:
I think the example _is_ using an environment variable, so perhaps remove this line?

Reply:
I had meant it as a warning: Look, do like we are doing, and please don't revert to the easy but insecure choice. But it does read like a disclaimer: Look, this is bad, please don't do this, I'm doing this only because it is convenient. Perhaps I could make that clearer.

Reply:
I'll pull this out, because we do cover the use of environment variables for security reasons in the Quickstart guide.